### ISSUE TITLE
Prepare for upcoming Seurat v5 release

### ISSUE THREAD

**Issue by saketkc** [JGASmits/AnanseSeurat#34]
I am opening this issue as a notification because `AnanseSeurat` is listed [here](https://cran.r-project.org/web/packages/Seurat/index.html) as a package that relies (depends/imports/suggests) on Seurat. As you may know, we recently released Seurat v5 as a beta in March of this year, with new updates for spatial, multimodal, and massively scalable analysis. For more information on updates and improvements, check out our website https://satijalab.org/seurat/.
We are now preparing to release Seurat v5 to CRAN, and plan to submit it on October 23rd. While we have tried our best to keep things backward-compatible, it is possible that updates to Seurat and SeuratObject might break your existing functionality or tests. We wanted to reach out before the new version is on CRAN, so that there's time to report issues/incompatibilities and prepare you for any changes in your code base that might be necessary.

We apologize for any disruption or inconvenience, but hope that the improvements to Seurat v5 will benefit your users going forward.
To test the upcoming release, you can install Seurat from the `seurat5` branch using the instructions available on this page: https://satijalab.org/seurat/articles/install.

Thank you!
Seurat v5 team

**Comment by saketkc** [comment]
Hi @JGASmits,

Seuratv5 will make its way to CRAN soon and reverse-dependency checks indicate following tests will fail:

```
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:

  ══ Failed tests ════════════════════════════════════════════════════════════════
  ── Error ('test-Motif2TF_functions.R:6:5'): Maelstrom_Motif2TF works ───────────
  Error in `t.default(exp_plot)`: argument is not a matrix
  Backtrace:
      ▆
   1. └─AnanseSeurat::Maelstrom_Motif2TF(...) at test-Motif2TF_functions.R:6:4
   2.   ├─base::t(scale(t(exp_plot)))
   3.   ├─base::scale(t(exp_plot))
   4.   ├─base::t(exp_plot)
   5.   └─base::t.default(exp_plot)

  [ FAIL 1 | WARN 1 | SKIP 0 | PASS 30 ]
```

One potential fix would be to do:
```r
op <- options(Seurat.object.assay.brackets = "v3")
on.exit(options(op), add = TRUE)
```
so that your users will stick to V3 assays. But it is possible you might have to update your accessor methods to be compatible with Seurat5 going ahead.

**Comment by JGASmits** [comment]
Dear Saketkc, 

The just released 1.2.0 version now supports Seurat V5, thanks for the clear communication about the release of Sereurat V5 and your patience. 

Greetings

JGA Smits
